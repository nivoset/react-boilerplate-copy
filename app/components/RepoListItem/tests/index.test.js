/**
 * Test the repo list item
 */

import React from 'react';
import { getByText, render } from '@testing-library/react';
import { IntlProvider } from 'react-intl';

import RepoListItem from '../index';

const renderComponent = item =>
  render(
    <IntlProvider locale="en">
      <RepoListItem item={item} />
    </IntlProvider>,
  );

describe('<RepoListItem />', () => {
  let ownedItem;
  let notOwnedItem;

  // Before each test reset the item data for safety
  beforeEach(() => {
    // Repo owned by the user whose account we're querying
    ownedItem = {
      isOwnRepo: true,
      owner: {
        login: 'repo-owner',
      },
      html_url: 'https://github.com/repo-owner/repo_name',
      name: 'repo_name',
      open_issues_count: 20,
      full_name: 'repo-owner/repo-name',
    };

    // Repo NOT owned by the user whose account we're querying
    notOwnedItem = {
      isOwnRepo: false,
      owner: {
        login: 'repo-owner',
      },
      html_url: 'https://github.com/repo-owner/repo_name',
      name: 'repo_name',
      open_issues_count: 20,
      full_name: 'repo-owner/repo-name',
    };
  });

  it('should render a ListItem', () => {
    const { container } = renderComponent(notOwnedItem);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should not render the username if repo is owned by user', () => {
    const { queryByText } = renderComponent(ownedItem);
    expect(queryByText(ownedItem.owner.login)).toBeNull();
  });

  it('should render username if the repo belongs to a differet user or org', () => {
    const { container } = renderComponent(notOwnedItem);
    expect(
      getByText(container, content =>
        content.startsWith(notOwnedItem.owner.login),
      ),
    ).not.toBeNull();
  });

  it('should render the repo name', () => {
    const { container } = renderComponent(ownedItem);
    expect(
      getByText(container, content => content.endsWith(ownedItem.name)),
    ).not.toBeNull();
  });

  it('should render the issue count', () => {
    const { container } = renderComponent(ownedItem);
    expect(
      getByText(container, ownedItem.open_issues_count.toString(10)),
    ).not.toBeNull();
  });

  it('should render the IssueIcon', () => {
    const { container } = renderComponent(ownedItem);
    expect(container.querySelector('svg')).not.toBeNull();
  });
});
