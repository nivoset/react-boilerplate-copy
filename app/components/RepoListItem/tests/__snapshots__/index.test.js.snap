// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepoListItem /> should render a ListItem 1`] = `
<li
  class="Wrapper-euo0oy-0 jLvxtc"
>
  <div
    class="Item-sc-3y9mie-0 fKXSkT"
  >
    <div
      class="Wrapper-sc-13xnchy-0 kxxMFK"
    >
      <a
        class="A-br8h0y-0 RepoLink-sc-1x99nq5-0 cUSWhT"
        href="https://github.com/repo-owner/repo_name"
        target="_blank"
      >
        repo-owner/repo_name
      </a>
      <a
        class="A-br8h0y-0 IssueLink-sc-16kygqm-0 giwGlR"
        href="https://github.com/repo-owner/repo_name/issues"
        target="_blank"
      >
        <svg
          class="IssueIcon-k8snxy-0 GsPBd"
          height="1em"
          width="0.875em"
        >
          <path
            d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7S10.14 13.7 7 13.7 1.3 11.14 1.3 8s2.56-5.7 5.7-5.7m0-1.3C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7S10.86 1 7 1z m1 3H6v5h2V4z m0 6H6v2h2V10z"
          />
        </svg>
        <span>
          20
        </span>
      </a>
    </div>
  </div>
</li>
`;
